components/sst_importer/src/sst_importer.rs

```diff
@@ -306,13 +310,27 @@ impl SstImporter {
             return Ok(path.save);
         }
 
+        if let Some(p) = path.temp.parent() {
```

Comment:
why move it outside works? can we have some test to cover this? in case other people change it in future

Reply:
If download file without KMS, the process will run the default interface ExternalStorage::restore() , in which the subdir have been created.
(https://github.com/tikv/tikv/blob/ba391ff506c8b7f4b0cd7c9ef0b9f04ce87c3d7e/components/external_storage/src/lib.rs#L80);  

If download file with KMS, it will run the EncryptedExternalStorage::restore, in which the subdir have not been created.
(https://github.com/tikv/tikv/blob/ba391ff506c8b7f4b0cd7c9ef0b9f04ce87c3d7e/components/external_storage/export/src/export.rs#L326)

After fix it, It will create subdir firstly, no matter which interface is called.